xarray/core/duck_array_ops.py

```diff
@@ -278,6 +276,25 @@ def f(values, axis=None, skipna=None, **kwargs):
 cumsum_1d.numeric_only = True
 
 
+_mean = _create_nan_agg_method('mean')
+
+
+def mean(array, axis=None, skipna=None, **kwargs):
+    """ inhouse mean that can handle datatime dtype """
+    array = asarray(array)
+    if array.dtype.kind == 'M':
+        offset = min(array)
+        # infer the compatible timedelta dtype
+        dtype = (np.empty((1,), dtype=array.dtype) - offset).dtype
```

Comment:
This is just to find the corresponding timedelta from datetime. Is there any good function to find an appropriate dtype?

Reply:
Thanks. I just realized we are always using [ns] for datetime. Updated.